src/main/java/com/algorand/algosdk/logic/Logic.java

```diff
@@ -147,28 +164,56 @@ public static VarintResult getUVarint(byte [] buffer, int bufferOffset) {
         return new VarintResult();
     }
 
+    public static void sanityCheckProgram(final byte[] program) {
```

Comment:
@ahangsu Leaving a visual reminder to revisit once we agree on `sanityCheckProgram` internals:  The method should provide javadoc to summarize usage to API users.